src/rpc/core_rpc_server_commands_defs.h

```diff
@@ -3415,4 +3415,112 @@ constexpr char const CORE_RPC_STATUS_TX_LONG_POLL_MAX_CONNECTIONS[] = "Daemon ma
+      crypto::hash txid;          // The txid of who purchased the mapping (only applicable to Lokinet and if the Lokinet entry previously existed), null hash if not applicable
+      crypto::hash prev_txid; // The previous txid that purchased the mapping (only applicable to Lokinet and if the Lokinet entry previously existed), null hash if not applicable
+      BEGIN_KV_SERIALIZE_MAP()
+        KV_SERIALIZE(entry_index)
+        KV_SERIALIZE(type)
+        KV_SERIALIZE(owner)
+        KV_SERIALIZE(pubkey)
+        KV_SERIALIZE(register_height)
+        KV_SERIALIZE_VAL_POD_AS_BLOB(txid)
+        KV_SERIALIZE_VAL_POD_AS_BLOB(prev_txid)
+      END_KV_SERIALIZE_MAP()
+    };
+
+    struct response
+    {
+      std::vector<response_entry> entries;
+      std::string status; // Generic RPC error code. "OK" is the success value.
+      BEGIN_KV_SERIALIZE_MAP()
+        KV_SERIALIZE(entries)
+        KV_SERIALIZE(status)
+      END_KV_SERIALIZE_MAP()
+    };
+  };
+
+  LOKI_RPC_DOC_INTROSPECT
+  // Get all the name mappings for the queried owner. The owner should a ED25519
+  // public key, by default this is the purchaser's wallet spend key used to
+  // seed an ED25519 key.
+  struct COMMAND_RPC_GET_LNS_OWNERS_TO_NAMES
+  {
+    static size_t const MAX_REQUEST_ENTRIES = 256;
+    struct request
+    {
+      std::vector<std::string> entries; // The ed25519 public key to find all Loki Name Service entries for.
+      BEGIN_KV_SERIALIZE_MAP()
+        KV_SERIALIZE(entries)
+      END_KV_SERIALIZE_MAP()
+    };
+
+    struct response_mapping
+    {
+      uint16_t     type;            // The cateogry the Loki Name Service entry belongs to (blockchain = 0, lokinet = 1, messenger = 2, otherwise custom)
+      std::string  name;            // The ed25519 public key that purchased the Loki Name Service entry.
+      std::string  pubkey;          // The public key that the name maps to.
+      uint64_t     register_height; // The height that this Loki Name Service entry was purchased on the Blockchain.
+      crypto::hash txid;            // The height that this Loki Name Service entry was purchased on the Blockchain.
+      crypto::hash prev_txid;   // The previous txid that purchased the mapping (only applicable to Lokinet and if the Lokinet entry previously existed), null hash if not applicable
+      BEGIN_KV_SERIALIZE_MAP()
+        KV_SERIALIZE(type)
+        KV_SERIALIZE(name)
+        KV_SERIALIZE(pubkey)
+        KV_SERIALIZE_VAL_POD_AS_BLOB(txid)
+        KV_SERIALIZE_VAL_POD_AS_BLOB(prev_txid)
+      END_KV_SERIALIZE_MAP()
+    };
+
+    struct response_entry
+    {
+      uint64_t entry_index;                   // The index in request_entry's `entries` array that was resolved via Loki Name Service.
+      std::vector<response_mapping> mappings;
+      BEGIN_KV_SERIALIZE_MAP()
+        KV_SERIALIZE(entry_index)
+        KV_SERIALIZE(mappings)
+      END_KV_SERIALIZE_MAP()
+    };
```

Comment:
Hm, you want me to add `owner` to the response_mapping? Otherwise you won't be able to link up the result to the originating request since an owner can result in many names.

Reply:
Can't I link it up by just using the position in the returned value (which is implicitly the "entry_index")?

Reply:
From out-of-band discussion: It seems that we can't do the nicer interface here because epee.